### ISSUE TITLE
[GR-45281] Native-Image complains about processor architecture even if it's OK

### ISSUE THREAD

**Issue by vinc3nz** [oracle/graal#6301]
```
F:\Vincenz\Developement>native-image Test
========================================================================================================================
GraalVM Native Image: Generating 'test' (executable)...
========================================================================================================================

[1/7] Initializing...                                                                                    (0,0s @ 0,08GB)
Error: Native-image building on Windows currently only supports target architecture: AMD64 (32-bit architecture x86 unsupported)
Error: To prevent native-toolchain checking provide command-line option -H:-CheckToolchain
Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
------------------------------------------------------------------------------------------------------------------------
                         0,2s (5,0% of total time) in 8 GCs | Peak RSS: 0,53GB | CPU load: 2,88
========================================================================================================================
Failed generating 'test' after 3,1s.
Error: Image build request failed with exit status 1

F:\Vincenz\Developement>set processor
PROCESSOR_ARCHITECTURE=AMD64
PROCESSOR_IDENTIFIER=AMD64 Family 23 Model 8 Stepping 2, AuthenticAMD
PROCESSOR_LEVEL=23
PROCESSOR_REVISION=0802

F:\Vincenz\Developement>java -version
java version "17.0.6" 2023-01-17 LTS
Java(TM) SE Runtime Environment GraalVM EE 22.3.1 (build 17.0.6+9-LTS-jvmci-22.3-b11)
Java HotSpot(TM) 64-Bit Server VM GraalVM EE 22.3.1 (build 17.0.6+9-LTS-jvmci-22.3-b11, mixed mode, sharing)

F:\Vincenz\Developement>
```

I have an AMD Ryzen 5 2600 on this PC.

**Comment by fniephaus** [comment]
Are you sure that your Visual Studio installation is for x64 and not for x86? What do you see when you run `cl.exe -v` in your Windows terminal?

**Comment by vinc3nz** [comment]
```
C:\Users\Vincenz>cl -v
Microsoft (R) C/C++-Optimierungscompiler Version 19.35.32216.1 für x64 // C/C++ optimization compiler ... for x64
Copyright (C) Microsoft Corporation. Alle Rechte vorbehalten. // All rights reserved

cl : Befehlszeile warning D9002 : Unbekannte Option "-v" wird ignoriert. // Ignoring unkown option "-v"
cl : Befehlszeile error D8003 : Name der Quelldatei fehlt. // Name of source file is missing

C:\Users\Vincenz>
```
Translated to englisgh for you

**Comment by fniephaus** [comment]
Thanks! It seems that the version check fails if your Windows environment is set to German. We're going to look into a fix for this. In the meantime, please use `-H:-CheckToolchain` as a workaround. Thanks for reporting!

**Comment by fniephaus** [comment]
It seems we've already addressed this issue a while ago, but the fix is not in 22.3.1. Could you please try a recent [dev build](https://github.com/graalvm/graalvm-ce-dev-builds/releases) and confirm that it's working for you?

**Comment by fniephaus** [comment]
We finally fixed this with https://github.com/oracle/graal/pull/8058.
